src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/TranscodingWriteStream.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Taken from https://github.com/dotnet/aspnetcore/blob/master/src/Mvc/Mvc.Core/src/Formatters/TranscodingWriteStream.cs
+
+using System.Buffers;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    internal sealed class TranscodingWriteStream : Stream
+    {
+        internal const int MaxCharBufferSize = 4096;
+        internal const int MaxByteBufferSize = 4 * MaxCharBufferSize;
```

Comment:
This decision comes from https://github.com/dotnet/aspnetcore/pull/12142#discussion_r304970713.

Reply:
There should at minimum be a comment explaining why the constant was chosen. Keep in mind that the max chars-to-bytes expansion for UTF-8 is 3 (not 4) bytes per char. But clearly we're not transcoding _to_ UTF-8 (since the input was already UTF-8), so using any UTF-8 specific constant would not be correct.

It's perfectly ok if we say something like "this is a reasonable guess for a worst-case expansion for a typical encoding". But IMO there needs to be a comment that says _something_. :)

Reply:
Done.
Added the following comment:
```
Upper bound that limits the byte buffer size to prevent an encoding that has a very poor worst-case scenario.
```

Now that makes me question if we should limit the char buffer on the Read side for the same reason.